### PR TITLE
Add vault name and symbol

### PR DESCRIPTION
**Pull request by ben-kaufman**

### CODE REVIEW

File: contracts/interfaces/IHATVaultsRegistry.sol [hunk 1]
```diff
@@ -284,6 +285,7 @@ interface IHATVaultsRegistry {
         address indexed _asset,
         address indexed _owner,
         address _committee,
+        string _symbol,
```
Comment (jellegerbrandy) [inline contracts/interfaces/IHATVaultsRegistry.sol:288]: why not add `_name` as well?

File: contracts/HATVaultsRegistry.sol [hunk 2]
```diff
@@ -264,52 +264,26 @@ contract HATVaultsRegistry is IHATVaultsRegistry, Ownable {
...
-        HATVault.BountySplit calldata _bountySplit,
-        string calldata _descriptionHash,
-        uint32 _bountyVestingDuration,
-        uint32 _bountyVestingPeriods,
-        bool _isPaused
-    ) 
-    external 
-    returns(address vault)
-    {
+    function createVault(IHATVault.VaultInitParams calldata _params) external returns(address vault) {
```
Comment (jellegerbrandy) [inline contracts/HATVaultsRegistry.sol:267]: actually i liked the separate arguments better though :)

Comment (ben-kaufman) [reply contracts/HATVaultsRegistry.sol:267]: Yeah but get stack too deep

File: contracts/interfaces/IHATVaultsRegistry.sol (hunk 1 above)
Comment (ben-kaufman) [reply contracts/interfaces/IHATVaultsRegistry.sol:288]: Well I didn't know if I should add it or derive from the symbol.

**Comment by jellegerbrandy** [comment]
Ok, I propose that symbol and name get prefixed as follows:

```
contructor(_symbol, _name) {
  symbol = "HATS" + symbol; // prefix with "HATS", no space, so we get "HATSETH", "HATSDXD", etc. 
 name = "HATs Vault: " + name; // prefix with "HATs Vault: ", so we get "HATs Vault: DXD bounty"
}
```

**Comment by jellegerbrandy** [comment]
but let's discuss in the call

File: contracts/HATVault.sol [hunk 3]
```diff
@@ -176,6 +176,7 @@ contract HATVault is IHATVault, ERC4626Upgradeable, OwnableUpgradeable, Reentran
         if (_params.maxBounty > MAX_BOUNTY_LIMIT)
             revert MaxBountyCannotBeMoreThanMaxBountyLimit();
         _validateSplit(_params.bountySplit);
+        __ERC20_init(string.concat("HATs Vault ", _params.name), string.concat("HAT", _params.symbol));
```
Comment (lirona) [inline contracts/HATVault.sol:179]:
```suggestion
        __ERC20_init(string.concat("Hats Vault ", _params.name), string.concat("HAT", _params.name));
```

Comment (lirona) [reply contracts/HATVault.sol:179]: the shares token symbol should also be with the name of the vault and not the name of the vault's token, e.g HATAngle, HATDXDAO, HATICHI.. and not HATUSDC

File: contracts/interfaces/IHATVault.sol [hunk 4]
```diff
@@ -73,6 +73,8 @@ interface IHATVault is IERC4626Upgradeable {
 
     /**
     * @notice Initialization parameters for the vault
+    * @param name The vault's name (concatinated as "HATs Vault " + name)
+    * @param symbol The vault's symbol (concatinated as "HAT" + symbol)
```
Comment (lirona) [inline contracts/interfaces/IHATVault.sol:77]:
```suggestion
    * @param symbol The vault's symbol (concatenated as "HAT" + name)
```

File: contracts/interfaces/IHATVault.sol [hunk 5]
```diff
@@ -73,6 +73,8 @@ interface IHATVault is IERC4626Upgradeable {
 
     /**
     * @notice Initialization parameters for the vault
+    * @param name The vault's name (concatinated as "HATs Vault " + name)
```
Comment (lirona) [inline contracts/interfaces/IHATVault.sol:76]:
```suggestion
    * @param name The vault's name (concatenated as "Hats Vault " + name)
```

File: contracts/interfaces/IHATVault.sol [hunk 6]
```diff
@@ -91,6 +93,8 @@ interface IHATVault is IERC4626Upgradeable {
     * @dev Needed to avoid a "stack too deep" error
     */
     struct VaultInitParams {
+        string name;
+        string symbol;
```
Comment (lirona) [inline contracts/interfaces/IHATVault.sol:97]:
```suggestion
```

File: contracts/mocks/PoolsManagerMock.sol [hunk 7]
```diff
@@ -20,16 +20,20 @@ contract VaultsManagerMock {
...
-                                _descriptionHash,
-                                _bountyVestingDuration,
-                                _bountyVestingPeriods,
-                                false);
+            address vault = _hatVaults.createVault(IHATVault.VaultInitParams({
+                                    asset: _assets[i],
+                                    owner: _hatVaults.owner(),
+                                    committee: _committee,
+                                    name: "VAULT",
+                                    symbol: "VLT",
```
Comment (lirona) [inline contracts/mocks/PoolsManagerMock.sol:28]:
```suggestion
```

File: test/common.js [hunk 8]
```diff
@@ -130,18 +130,20 @@ const setup = async function(
...
-    86400,
-    10,
-    false
-  )).logs[1].args._vault);
+  let vault = await HATVault.at((await hatVaultsRegistry.createVault({
+    asset: stakingToken.address,
+    owner: await hatVaultsRegistry.owner(),
+    committee: accounts[1],
+    name: "VAULT",
+    symbol: "VLT",
```
Comment (lirona) [inline test/common.js:138]:
```suggestion
```

File: test/hattimelockcontroller.js [hunk 9]
```diff
@@ -91,18 +91,20 @@ const setup = async function(
...
-    86400,
-    10,
-    false
-  )).receipt.rawLogs[0].address);
+  vault = await HATVault.at((await hatVaultsRegistry.createVault({
+    asset: stakingToken.address,
+    owner: await hatVaultsRegistry.owner(),
+    committee: accounts[1],
+    name: "VAULT",
+    symbol: "VLT",
```
Comment (lirona) [inline test/hattimelockcontroller.js:99]:
```suggestion
```

File: test/hattimelockcontroller.js [hunk 10]
```diff
@@ -440,18 +442,20 @@ contract("HatTimelockController", (accounts) => {
...
-      86400,
-      10,
-      false
-    )).receipt.rawLogs[0].address);
+    let newVault = await HATVault.at((await hatVaultsRegistry.createVault({
+        asset: stakingToken2.address,
+        owner: await hatVaultsRegistry.owner(),
+        committee: accounts[3],
+        name: "VAULT",
+        symbol: "VLT",
```
Comment (lirona) [inline test/hattimelockcontroller.js:450]:
```suggestion
```

File: test/hatvaults.js [hunk 11]
```diff
@@ -576,36 +576,40 @@ contract("HatVaults", (accounts) => {
...
-      86400,
-      10,
-      false
-    );
+    let tx = await hatVaultsRegistry.createVault({
+      asset: stakingToken2.address,
+      owner: await hatVaultsRegistry.owner(),
+      committee: accounts[3],
+      name: "VAULT",
+      symbol: "VLT",
```
Comment (lirona) [inline test/hatvaults.js:584]:
```suggestion
```

File: test/hatvaults.js [hunk 12]
```diff
@@ -576,36 +576,40 @@ contract("HatVaults", (accounts) => {
...
-    assert.equal(tx.logs[1].args._bountySplit.committee, "500");
-    assert.equal(tx.logs[1].args._descriptionHash, "_descriptionHash1");
-    assert.equal(tx.logs[1].args._bountyVestingDuration, 86400);
-    assert.equal(tx.logs[1].args._bountyVestingPeriods, 10);
-    assert.equal(tx.logs[1].args._isPaused, false);
+    assert.equal(tx.logs[1].args._params.asset, stakingToken2.address);
+    assert.equal(tx.logs[1].args._params.owner, await hatVaultsRegistry.owner());
+    assert.equal(tx.logs[1].args._params.committee, accounts[3]);
+    assert.equal(tx.logs[1].args._params.name, "VAULT");
+    assert.equal(tx.logs[1].args._params.symbol, "VLT");
```
Comment (lirona) [inline test/hatvaults.js:603]:
```suggestion
```

File: test/hatvaults.js [hunk 13]
```diff
@@ -614,11 +618,17 @@ contract("HatVaults", (accounts) => {
     assert.equal(logs[0].event, "SetVaultDescription");
     assert.equal(logs[0].args._descriptionHash, "_descriptionHash1");
 
+    let vault = await HATVault.at(tx.logs[1].args._vault);
+    assert.equal(await vault.name(), "HATs Vault VAULT");
+    assert.equal(await vault.symbol(), "HATVLT");
```
Comment (lirona) [inline test/hatvaults.js:623]:
```suggestion
```

File: test/hatvaults.js [hunk 14]
```diff
@@ -614,11 +618,17 @@ contract("HatVaults", (accounts) => {
...
+
     try {
       await vault.initialize({
         asset: stakingToken.address,
         owner: await hatVaultsRegistry.owner(),
         committee: accounts[1],
+        name: "VAULT",
+        symbol: "VLT",
```
Comment (lirona) [inline test/hatvaults.js:631]:
```suggestion
```

File: test/hatvaults.js [hunk 15]
```diff
@@ -652,18 +662,20 @@ contract("HatVaults", (accounts) => {
...
-      86400,
-      10,
-      false
-    )).logs[1].args._vault);
+    let newVault = await HATVault.at((await hatVaultsRegistry.createVault({
+      asset: stakingToken2.address,
+      owner: await hatVaultsRegistry.owner(),
+      committee: accounts[3],
+      name: "VAULT",
+      symbol: "VLT",
```
Comment (lirona) [inline test/hatvaults.js:670]: and remove all `symbol`... :)

**Review by lirona — COMMENTED** [review]
looks good, just one change - if i understood our decision it was that the symbol derived would be with the vault's name and not token symbol

File: contracts/HATVault.sol (hunk 3 above)
Comment (jellegerbrandy) [reply contracts/HATVault.sol:179]: mm, as i understood our decision was to provide two separate arguments, for _symbol and _name (and so we do not change this)

File: contracts/interfaces/IHATVault.sol (hunk 4 above)
Comment (ben-kaufman) [reply contracts/interfaces/IHATVault.sol:77]: Why? they are different the name and symbol and we pass both

File: contracts/interfaces/IHATVault.sol (hunk 5 above)
Comment (ben-kaufman) [reply contracts/interfaces/IHATVault.sol:76]: But it is "HATs Vault", do we want to change this capitalization?

File: contracts/HATVault.sol (hunk 3 above)
Comment (ben-kaufman) [reply contracts/HATVault.sol:179]: I also understood we want to pass name and symbol, we can't really derive the symbol from the name

Comment (lirona) [reply contracts/HATVault.sol:179]: the symbol is HAT+the project name, isn't it?

File: contracts/interfaces/IHATVault.sol (hunk 5 above)
Comment (lirona) [reply contracts/interfaces/IHATVault.sol]: yes keep this, as we discussed :)

File: contracts/interfaces/IHATVault.sol (hunk 4 above)
Comment (lirona) [reply contracts/interfaces/IHATVault.sol]: right, so just the typo

**Review by jellegerbrandy — APPROVED** [review]

**Pull request merged**